evalml/data_checks/target_leakage_data_check.py

```diff
@@ -13,23 +13,49 @@
 class TargetLeakageDataCheck(DataCheck):
     """Check if any of the features are highly correlated with the target."""
 
-    def __init__(self, pct_corr_threshold=0.95):
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
         """Check if any of the features are highly correlated with the target.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, supports all target and feature types. Otherwise, only supports binary with numeric and boolean dtypes.
+        Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
 
         Arguments:
             pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
+            pearson_corr (bool): Whether or not to use the Pearson Correlation versus mutual information. Defaults to False, which uses mutual info.
 
         """
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
         self.pct_corr_threshold = pct_corr_threshold
+        self.pearson = pearson_corr
+
+    def _calculate_pearson(self, X, y, messages):
+        highly_corr_cols = []
+        X = X.select_dtypes(include=numeric_and_boolean_dtypes)
+        if len(X.columns) > 0:
+            highly_corr_cols = [label for label, col in X.iteritems() if abs(y.corr(col)) >= self.pct_corr_threshold]
+        return highly_corr_cols
+
+    def _calculate_mutual_information(self, X, y):
+        highly_corr_cols = []
+        target = 'target'
+        while target in X.columns:
+            target += '0'
```

Comment:
I guess this is for my edification: is "target" always guaranteed to be in the incoming data frame?  I'm a little wary on there being a hardcoded check on it.

Reply:
@chukarsten, 'target' isn't guaranteed, but this code allows me to safely add the `y` data to the `X` dataframe in order to find mutual information. The default name for column with `y` would be `target`, but if `target` already exists as a column name, I wanted to find another name to allow the data to be safely added without replacing/overwriting any other columns